pkg/restapi/vc/operation/operations.go

```diff
@@ -1490,13 +1499,57 @@ func (o *Operation) retrieveVC(profileName, docID, contextErrText string) ([]byt
 	return retrievedVC, nil
 }
 
-func getPublicKeyID(didDoc *ariesdid.Doc) (string, error) {
+func (o *Operation) createPublicKey(pr *ProfileRequest) (*didclient.PublicKey, string, error) {
+	var didPrivateKey string
+
+	var publicKey *didclient.PublicKey
+
+	switch pr.DIDKeyType {
+	case "", crypto.Ed25519KeyType:
+		_, base58PubKey, err := o.kms.CreateKeySet()
+		if err != nil {
+			return nil, "", err
+		}
+
+		publicKey = &didclient.PublicKey{ID: pubKey1, Type: didclient.JWSVerificationKey2020,
+			Value: base58.Decode(base58PubKey), Encoding: didclient.PublicKeyEncodingJwk,
+			KeyType: didclient.Ed25519KeyType, Usage: []string{didclient.KeyUsageGeneral, didclient.KeyUsageOps}}
+	case crypto.P256KeyType:
+		privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+		if err != nil {
+			return nil, "", err
+		}
+
+		pubKeyBytes, err := x509.MarshalPKIXPublicKey(privateKey.Public())
+		if err != nil {
+			return nil, "", err
+		}
+
+		encodedPrivateKey, err := x509.MarshalECPrivateKey(privateKey)
+		if err != nil {
+			return nil, "", err
+		}
+
+		// save the private key in DID as KMS only supports ed25519 keys
+		didPrivateKey = base58.Encode(encodedPrivateKey)
+
+		publicKey = &didclient.PublicKey{ID: pubKey1, Type: didclient.JWSVerificationKey2020,
```

Comment:
why we just add two keys one for P256KeyType and other for Ed25519KeyType

Reply:
pubKey1 for Ed25519KeyType.  pubKey2 for ECKeyType

Reply:
You mean always add 2 keys ?

Reply:
yes

Reply:
we can change it later. but the plan that we need did doc to include multiple keys

Reply:
yeh we should also show DID with multiple keys.

Reply:
created an issue https://github.com/trustbloc/edge-service/issues/300